addon/components/-focusable.js

```diff
@@ -0,0 +1,175 @@
+/**
+ * @module ember-paper
+ */
+import Component from '@glimmer/component';
+import { tracked } from '@glimmer/tracking';
+import { action } from '@ember/object';
+
+/**
+ * @class Focusable
+ * @extends @glimmer/component
+ *
+ * When extending from Focusable it is expected that md-focused be implemented
+ * on the top level tag along with setting tabindex and disabled. This component
+ * listens to a large number of events, therefore render listener register
+ * functions have been created to ease usage. Clearly, this is non-optimal and
+ * only the listeners that are required should be added using the `on` modifier.
+ *
+ * Use the following as a base:
+ * ```hbs
+ * <myTag class='{{if this.focused " md-focused"}}' disabled={{this.disabled}} tabindex={{if this.disabled "-1" "0"}} {{did-insert this.registerListeners}} {{will-destroy this.unregisterListeners}} ...attributes>
+ * </myTag>
+ * ```
+ */
+export default class Focusable extends Component {
+  @tracked pressed = false;
+  @tracked active = false;
+  @tracked focused = false;
+  @tracked hover = false;
+
+  // classNameBindings: ['focused:md-focused'],
+  // attributeBindings: ['tabindex', 'disabledAttr:disabled'],
+
+  get disabled() {
+    return this.args.disabled || false;
+  }
+
+  toggle = false;
+
+  // Only render the "focused" state if the element gains focus due to
+  // keyboard navigation.
+  get focusOnlyOnKey() {
+    return this.args.focusOnlyOnKey ? this.args.focusOnlyOnKey : false;
+  }
+
+  @action registerListeners(element) {
```

Comment:
this _feels_ like an insane amount of listeners... I think a lot of it has to do with animating ripple effects